Options.py

```diff
@@ -854,13 +855,51 @@ def __iter__(self) -> typing.Iterator[str]:
     def __len__(self) -> int:
         return self.value.__len__()
 
+    # __getitem__ fallback fails for Counters, so we define this explicitly
+    def __contains__(self, item):
+        return item in self.value
+
+
+class CounterOption(OptionDict):
+    min: typing.Optional[int] = None
+    max: typing.Optional[int] = None
+
+    def __init__(self, value: typing.Dict[str, int]):
+        super(CounterOption, self).__init__(collections.Counter(value))
+
+    def verify(self, world: typing.Type[World], player_name: str, plando_options: PlandoOptions) -> None:
+        super(CounterOption, self).verify(world, player_name, plando_options)
 
-class ItemDict(OptionDict):
+        range_errors = []
+
+        if self.max is not None:
+            range_errors += [
+                f"\"{key}: {value}\" is higher than maximum allowed value {self.max}."
+                for key, value in self.value.items() if value > self.max
+            ]
+
+        if self.min is not None:
+            range_errors += [
+                f"\"{key}: {value}\" is lower than minimum allowed value {self.min}."
+                for key, value in self.value.items() if value < self.min
+            ]
+
+        if len(range_errors) == 1:
+            raise OptionError(range_errors[0][:-1] + f" for option {getattr(self, 'display_name', self)}.")
+        elif range_errors:
+            range_errors = [f"For option {getattr(self, 'display_name', self)}:"] + range_errors
+            raise OptionError("\n".join(range_errors))
+
+
+class ItemDict(CounterOption):
     verify_item_name = True
 
+    min = 0
+
     def __init__(self, value: typing.Dict[str, int]):
-        if any(item_count < 1 for item_count in value.values()):
-            raise Exception("Cannot have non-positive item counts.")
+        # Backwards compatibility: Cull 0s to make "in" checks behave the same as when this wasn't a CounterOption
```

Comment:
Since this is just for back compat and verify is called after the option itself is initialized, min should probably be 1 and just leave the culling.

Separate conversation ig but it seems weird to leave in the culling at all since it's not like the actual option was getting those keys with 0 Val already

Reply:
Ohh, hm, I definitely intended that for start_inventory, `{"Sword": 0}` is a valid input now.